EIPS/eip-7110.md

```diff
@@ -0,0 +1,213 @@
+
+        // Return the final owner in the chain
+        return currentOwnerAddress;
+    }
+
+    // This internal function is used to implement the transfer of tokens, following the ERC721 standard but allowing dynamic token ownership.
+    // It transfers the `tokenId` token from the `from` address to the `to` address.
+    function _transfer(address from, address to, uint256 tokenId) internal virtual override {
+        require(ownerOf(tokenId) == from, "ERC721D: transfer of token that is not owned");
+        Ownership memory oldOwnership = _owners[tokenId];
+        
+        _approve(address(0), tokenId);
+        _owners[tokenId] = Ownership(to, 0);
+
+        if (oldOwnership.ownerAddress == address(0)) {
+            // The token is being owned for the first time, increase the balance of the new owner
+            _balances[to] += 1;
+        } else if (oldOwnership.ownerAddress != to) {
+            // The token is changing owner, adjust the balances
+            address oldOwner = oldOwnership.ownerAddress;
+            _balances[oldOwner] -= 1;
+            _balances[to] += 1;
+        }
+        
+        emit Transfer(from, to, tokenId);
+    }
+
+    // An internal function that checks if a `spender` is an approved operator or the owner of a token.
+    // Returns true if the `spender` is an approved operator or the owner of the `tokenId` token.
+    // The function follows the ERC721 standard requirements.
+    function _isApprovedOrOwner(address spender, uint256 tokenId) internal view override returns (bool) {
+        require(_exists(tokenId), "ERC721D: operator query for nonexistent token");
+        address owner = ownerOf(tokenId);
+        return (spender == owner || getApproved(tokenId) == spender || isApprovedForAll(owner, spender));
+    }
+
+    // Overrides the `balanceOf` function from the ERC721 standard.
+    // Returns the balance (number of owned tokens) of the `owner` address.
+    // It checks for the zero address and returns the balance from the internal _balances mapping.
+    function balanceOf(address owner) public view override(ERC721) returns (uint256) {
+        require(owner != address(0), "ERC721D: balance query for the zero address");
+        return _balances[owner];
+    }
+
+    // This function returns the ownership details of the `tokenId` token.
+    // Returns a struct with the owner's address and the token id of the token owned by the returned token (if any).
+    function owners(uint256 tokenId) public view returns (Ownership memory) {
+        return _owners[tokenId];
+    }
+
+}
+```
+
+## Rationale
+
+The `ERC-721D` standard seeks to expand the potential of NFTs by introducing dynamic ownership. This innovation could open up new use cases in the fields of digital assets, dApps, digital identity, and more. As the digital economy evolves, the need for complex and dynamic relationships between tokens will become increasingly relevant, and the `ERC-721D` standard addresses this need.
```

Comment:
This reads more like motivation. The Rationale section should justify technical choices made within the EIP itself, while the Motivation section justifies the EIP as a whole.

An example of something you _may_ want to explain in the Rationale section is why you chose EIP-721 as the base of your extension and not EIP-1155.